lib/extensions/ar_virtual.rb

```diff
@@ -201,11 +201,31 @@ def reflection_with_virtual(association)
     end
 
     def follow_associations(association_names)
-      association_names.inject(self) { |klass, name| klass.try!(:reflect_on_association, name).try!(:klass) }
+      collect_associations(association_names).try!(:last).try!(:klass)
     end
 
     def follow_associations_with_virtual(association_names)
-      association_names.inject(self) { |klass, name| klass.try!(:reflection_with_virtual, name).try!(:klass) }
+      collect_associations_with_virtual(association_names).try!(:last).try!(:klass)
+    end
+
+    def collect_associations(association_names)
```

Comment:
Just one thought - I'd expect `collect_associations` to return the reflection classes - WDYT about `collect_reflections`?